### PR TITLE
Add support for gradient clipping

### PR DESCRIPTION
**Pull request by hturki**
I've found that this makes training more stable in certain cases, especially when using high learning rates

### CODE REVIEW

**Review by tancik — APPROVED** [review]
LGTM, are you also planning to update the nerfacto config?

**Comment by hturki** [comment]
i wasn't planning on it, have been testing on fairly specific usecases so far with a hash-based (but not nerfacto) architecture, and I'm not sure whether this is needed for nerfacto / the specific clipping values I'm using will generalize. For nerfacto I'd maybe start with adding weight decay per https://github.com/nerfstudio-project/nerfstudio/issues/873 and then we could figure out reasonable clipping values if training is still unstable.

**Comment by tancik** [comment]
Gotcha 👍
